File: src/alphabet.ts
export { Alphabet };

import { Random } from "./random";
import { Glyph } from ".";

const numberOfLettersInAlphabet = 26;

class Alphabet {
  glyphs: any;

  constructor(private random: Random) {
    // let letters = new Array<Glyph>(numberOfLettersInAlphabet);
    this.glyphs = {};
    for (
      let currentLetter = 0;
      currentLetter < numberOfLettersInAlphabet;
      currentLetter++
    ) {
      this.glyphs[String.fromCharCode(65 + currentLetter)] = new Glyph(random);
    }
  }

  /**
   * Glyph
   */
  public Glyph(letter: string): Glyph {
    if (letter.toUpperCase() in this.glyphs) {
      return this.glyphs[letter.toUpperCase()];
    }
    return null;
  }
}

File: UnityExample/Assets/Transmutation/Scripts/Alphabet.cs
using UnityEngine;
using System.Collections.Generic;

namespace EliCDavis.Transmutation
{
    public class Alphabet
    {
        private const int numberOfLettersInAlphabet = 26;

        Dictionary<char, Glyph> glyphs;

        public Alphabet()
        {
            glyphs = new Dictionary<char, Glyph>();
            for (int i = 0; i < numberOfLettersInAlphabet; i++)
            {
                this.glyphs[(char)(65 + i)] = new Glyph();
            }
        }

        public Glyph Glyph(char letter)
        {
            if (glyphs.ContainsKey(letter))
            {
                return glyphs[letter];
            }
            return null;
        }

    }

}